queue/src/fiskaltrust.Middleware.Localization.QueueAT/Services/ExportService.cs

```diff
@@ -37,8 +38,7 @@ public async Task PerformRksvJournalExportAsync(long fromTimestamp, long toTimes
                 foreach (var queue in await _configurationRepository.GetQueueATListAsync())
                 {
                     // TODO: Implement DB calls to increase performance
-                    var journals = (await _journalATRepository.GetAsync())
-                        .Where(x => x.TimeStamp >= fromTimestamp && x.TimeStamp <= toTimestamp);
+                    var journals = _journalATRepository.GetByTimeStampRangeAsync(fromTimestamp, toTimestamp).ToListAsync().Result;
```

Comment:
Let's not use `Result` on an async method if we don't have to - instead, it would be better to do this more memory-efficiently. 

Since we need the content twice, it would be most efficient to manually run over this in an async foreach loop, could look like this:

```cs
var sscdIds = new HashSet<Guid>();
var receipts = new List<string>();

var journals = _journalATRepository.GetByTimeStampRangeAsync(fromTimestamp, toTimestamp);
await foreach (var journal in journals.OrderBy(x => x.Number)) 
{
  if (journal.ftSignaturCreationUnitId != Guid.Empty) 
  {
    sscds.Add(journal.ftSignaturCreationUnitId);
  }
  receipts.Add($"{journal.JWSHeaderBase64url}.{journal.JWSPayloadBase64url}.{journal.JWSSignatureBase64url}");
}

```

Reply:
There's obviously more room for improvement by e.g. also not loading the receipt strings into memory; however, let's see how far we come with this before we refactor this to be fully async.